caom2-datalink-server/src/main/java/ca/nrc/cadc/caom2/datalink/LinkQueryRunner.java

```diff
@@ -158,10 +159,12 @@ public void run()
 
     private void doIt()
     {
-
+        final String checkpointID = ParameterUtil.findParameterValue("ID", job.getParameterList()) + " - doIt";
+        final Profiler outerProfiler = new Profiler(LinkQueryRunner.class);
```

Comment:
The profilers are supposed to be reused, so I think you can get away with only one instead of three.